node/src/store/redb_store.rs

```diff
@@ -369,6 +347,25 @@ impl Store for RedbStore {
         self.get_by_height(height).await
     }
 
+    async fn wait_new_head(&self) -> u64 {
```

Comment:
With this fn added, would it make sense to express `wait_height` with a common code in `Store` using it? Seems like an opportunity to reduce code duplication.

Reply:
With forward syncing we could do that, however an edge case exists in backward syncing:

When the store is empty, we don't know the network head, so we don't know which height we should await for. Of course we can workaround this with a loop of `head_height` and a sleep, until head is initialized. I'm ok with both approaches.